### PR TITLE
Implemented new Imperial Guard skills

### PR DESCRIPTION
**Pull request by Haydrich**
* **Addressed Issue(s)**: 

* **Server Mode**: RE

* **Description of Pull Request**: 
This is the initial release of the new imperial guard skills, [Fixed casting time/Variable Casting time/Cooldown/Global Cool Time] are dummies until providing a new information relative.
I followed the information of 
https://ro.gnjoy.com/news/devnote/View.asp?category=1&seq=4184991&curpage=1
Translated: https://www.divine-pride.net/forum/index.php?/topic/7814-kro-27560-level-expansion-and-new-skills/
PS: These skills require new clientdate above 2024/08 depending on the release date from Ragnarok.

### CODE REVIEW

File: src/map/skill.cpp [hunk 1]
```diff
@@ -8659,7 +8662,7 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 		clif_skill_nodamage(src,*bl,skill_id,skill_lv);
 		i = map_foreachinrange(skill_area_sub, bl, skill_get_splash(skill_id, skill_lv), starget,
 				src, skill_id, skill_lv, tick, flag|BCT_ENEMY|SD_SPLASH|1, skill_castend_damage_id);
-		if( !i && ( skill_id == RK_WINDCUTTER || skill_id == NC_AXETORNADO || skill_id == LG_CANNONSPEAR || skill_id == SR_SKYNETBLOW || skill_id == KO_HAPPOKUNAI ) )
+		if( !i && ( skill_id == RK_WINDCUTTER || skill_id == NC_AXETORNADO || skill_id == LG_CANNONSPEAR || skill_id == SR_SKYNETBLOW || skill_id == KO_HAPPOKUNAI || skill_id == SR_RAMPAGEBLASTER ) )
```
Comment (Haydrich) [inline src/map/skill.cpp:8665]: SR_RAMPAGEBLASTER in kRO show the skill effect when there is no target, which is required.

**Comment by OptimusM** [comment]
hey hey ~ Thank you for this, but, you probably need to specify wich client date is needed to have this, otherwise people will complain or ask you to "sir ples fix it"

**Comment by Haydrich** [comment]
> hey hey ~ Thank you for this, but, you probably need to specify wich client date is needed to have this, otherwise people will complain or ask you to "sir ples fix it"

Done

File: db/re/skill_db.yml [hunk 2]
```diff
@@ -43010,6 +43010,97 @@ Body:
...
+        Area: 2
+      - Level: 8
+        Area: 2
+      - Level: 9
+        Area: 3
+      - Level: 10
+        Area: 3
+    CastCancel: true
+    Cooldown: 1500 #TODO: confirm cooldown
+    FixedCastTime: 500 #TODO: confirm fixed cast time
```
Comment (Atemo) [inline db/re/skill_db.yml:43050]:
```suggestion
    CastTime: 1000
    AfterCastActDelay: 1200
    Cooldown: 700
    FixedCastTime: 1000
```

File: db/re/skill_db.yml [hunk 3]
```diff
@@ -43010,6 +43010,97 @@ Body:
...
+    MaxLevel: 5
+    Type: Weapon
+    TargetType: Attack
+    Range: 2
+    Hit: Multi_Hit
+    HitCount: 3
+    Element: Weapon
+    GiveAp: 2
+    CastCancel: true
+    Cooldown: 1500 #TODO: confirm cooldown
```
Comment (Atemo) [inline db/re/skill_db.yml:43089]:
```suggestion
    AfterCastActDelay: 700
    Cooldown: 400
```

File: src/map/skill.cpp (hunk 1 above)
Comment (Atemo) [reply src/map/skill.cpp:8665]: Please open another PR if it is not related to the new skill

File: src/map/battle.cpp [hunk 4]
```diff
@@ -3016,6 +3016,7 @@ static bool is_attack_critical(struct Damage* wd, struct block_list *src, struct
 #ifdef RENEWAL
 			case ASC_BREAKER:
 #endif
+			case IG_RADIANT_SPEAR:
```
Comment (datawulf) [inline src/map/battle.cpp:3019]:
```suggestion
```
crit chance should not be halved

Comment (Lemongrass3110) [reply src/map/battle.cpp]: @datawulf are you sure about that?
From the skill description: "The effectiveness of critical modifier is applied by half."

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/battle.cpp (hunk 4 above)
Comment (Haydrich) [reply src/map/battle.cpp]: > @datawulf are you sure about that? From the skill description: "The effectiveness of critical modifier is applied by half."

He is right, what you mentioned is the default critical factor from kRO, that the critical damages is halved

**Pull request merged**